mne/io/egi/egimff.py

```diff
@@ -68,10 +70,47 @@ def _read_mff_header(filepath):
     if n_chans != summaryinfo['n_channels']:
         print("Error. Should never occur.")
 
-    info_filepath = filepath + "/" + "info.xml"  # add with filepath
+    # Check presence of PNS data
+    if 'PNS' in all_files:
+        pns_fpath = op.join(filepath, all_files['PNS']['signal'])
+        pns_blocks = _get_blocks(pns_fpath)
+
+        pns_file = op.join(filepath, 'pnsSet.xml')
+        pns_obj = parse(pns_file)
+        sensors = pns_obj.getElementsByTagName('sensor')
+        pns_names = []
+        pns_types = []
+        pns_units = []
+        for sensor in sensors:
+            sn = sensor.getElementsByTagName('number')[0].firstChild.data
+            name = sensor.getElementsByTagName('name')[0].firstChild.data
+            unit_elem = sensor.getElementsByTagName('unit')[0].firstChild
+            unit = ''
+            if unit_elem is not None:
+                unit = unit_elem.data
+
+            if name == 'ECG':
+                ch_type = 'ecg'
+            elif 'EMG' in name:
+                ch_type = 'emg'
+            else:
+                ch_type = 'bio'
+            pns_types.append(ch_type)
+            pns_units.append(unit)
+            pns_names.append(name)
+
+        summaryinfo.update(pns_types=pns_types, pns_units=pns_units,
+                           pns_names=pns_names, n_pns_channels=len(pns_names),
+                           pns_fname=all_files['PNS']['signal'],
+                           pns_sample_blocks=pns_blocks)
+
+    info_filepath = op.join(filepath, 'info.xml')  # add with filepath
     tags = ['mffVersion', 'recordTime']
     version_and_date = _extract(tags, filepath=info_filepath)
-    summaryinfo.update(version=version_and_date['mffVersion'][0],
+    version = ""
+    if len(version_and_date['mffVersion']):
+        version = version_and_date['mffVersion'][0]
```

Comment:
it means we can close #4574 right?

Reply:
Indeed It adresses the same issues and includes those fixes too.